### DISCUSSION TITLE
Coupler length in the component ring_crow()

### DISCUSSION THREAD

**Discussion by manoranjanminz** [gdsfactory/gdsfactory#2571] [Q&A]
Hi,

I wanted to change the lenght_x of the coupling sections in the CROW structure. However, the component ring_crow() does not take 'length_x' as an argument. It shoots an error message.

`c = crow_block_comp << gf.components.ring_crow(gaps=[0.4,0.5,0.5,0.5,0.5,0.5,0.4],radius=[100*um]*6, bends= [bend_circular]*6,input_straight_cross_section = x_strip_lig_1800nm, output_straight_cross_section = x_strip_lig_1800nm, ring_cross_sections = [x_strip_lig_1800nm]*6, length_x = 75)`

![image](https://github.com/gdsfactory/gdsfactory/assets/102401390/748771ab-efac-49ae-bef1-b6c761579541)

How do I access length_x in ring_crow()?

**Comment by joamatab** [comment]
how about you copy the code and create your own version?

if your version ends up being nicer you can (if you want) contribute it back

**Reply by manoranjanminz** [reply]
Thanks @joamatab ,

I did the same thing. I modified the code and added another argument 'length_x' in ring_crow(). Made few changes and it worked fine.

![image](https://github.com/gdsfactory/gdsfactory/assets/102401390/a65f2ac5-9dc2-4124-998b-9f3ced64c097)


I will add "length_y" as well and try to contribute.

**Reply by joamatab** [reply]
Looks great 😊
